scripts/provisioning/configcmd.py

```diff
@@ -30,11 +33,119 @@ def __init__(self, config: str):
+        retval = self.configure_haproxy()
+      except Exception as e:
+        sys.stderr.write(f'In self.configure_haproxy(), exception: {e}\n')
+        return 1
+      if retval == 0:
+        sys.stdout.write("INFO: Successfully configured haproxy on the node.\n")
+      else:
+        sys.stderr.write("INFO: Failed to configure haproxy on the node.\n")
+        return retval
+
+    else:
+      sys.stderr.write("INFO: Failed to configure openldap on the node.\n")
+
     return retval
+
+  def configure_openldap(self):
+    """Install and Configure Openldap over Non-SSL."""
+    # 1. Install and Configure Openldap over Non-SSL.
+    # 2. Enable slapd logging in rsyslog config
+    # 3. Set openldap-replication
+    # 4. Check number of nodes in the cluster
+    cmd = ['/opt/seagate/cortx/s3/install/ldap/setup_ldap.sh',
+           '--ldapadminpasswd',
+           f'{self.ldap_passwd}',
+           '--rootdnpasswd', 
+           f'{self.rootdn_passwd}',
+           '--skipssl']
+    handler = SimpleProcess(cmd)
+    stdout, stderr, retcode = handler.run()
+    if retcode != 0:
+      sys.stderr.write(f"{cmd} failed with err: {stderr}, out: {stdout}, ret: {retcode}\n")
+      return 1
+
+    if os.path.isfile("/opt/seagate/cortx/s3/install/ldap/rsyslog.d/slapdlog.conf"):
+      try:
+        os.makedirs("/etc/rsyslog.d")
+      except OSError as e:
+        if e.errno != errno.EEXIST:
+          sys.stderr.write(f"mkdir /etc/rsyslog.d failed with errno: {e.errno}, exception: {e}\n")
+          return 1
+      shutil.copy('/opt/seagate/cortx/s3/install/ldap/rsyslog.d/slapdlog.conf',
+                  '/etc/rsyslog.d/slapdlog.conf')
+
+    cmd = ['systemctl', 'restart', 'rsyslog']
+    handler = SimpleProcess(cmd)
+    stdout, stderr, retcode = handler.run()
+    if retcode != 0:
+      sys.stderr.write(f"{cmd} failed with err: {stderr}, out: {stdout}, ret: {retcode}\n")
+      return 1
+
+    if self.server_nodes_count < 2:
+      sys.stdout.write(f"INFO: Not setting ldap-replication as the cluster node_count {self.server_nodes_count} is less than 2.\n")
+      return 0
+
+    sys.stdout.write(f"INFO: Setting ldap-replication as the cluster node_count {self.server_nodes_count} is greater than 2.\n")
+    with open("hosts_list_file.txt", "w") as f:
+      for host in self.hosts_list:
+        f.write(f"{host}\n")
+    sys.stdout.write("setting ldap-replication on all cluster nodes..\n")
+    cmd = ['/opt/seagate/cortx/s3/install/ldap/replication/setupReplicationScript.sh',
+           '-h',
+           'hosts_list_file.txt',
+           '-p', 
+           f'{self.rootdn_passwd}']
+    handler = SimpleProcess(cmd)
+    stdout, stderr, retcode = handler.run()
+    if retcode != 0:
+      sys.stderr.write(f"{cmd} failed with err: {stderr}, out: {stdout}, ret: {retcode}\n")
+      return 1
+    os.remove("hosts_list_file.txt")
+
+    return 0
+
+  def configure_haproxy(self):
+    """Configure haproxy service."""
+    cmd = ['s3haproxyconfig',
```

Comment:
We should not call s3haproxyconfig binary, instead us haproxy class package. S3Haproxyconfig is also in python.

Reply:
Following the discussions with @amitwac2608  and @shahidSTX , it seems we need to update ./scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py and remove it from an executable, and restructure the current code and put it into a new file under scripts/provisioning/. The restructuring will also address modularization of the existing large run() function in s3_haproxy_config.py. This significantly affects the scope of work of the current ticket.

@AjinkyaDhumal  please suggest if this needs to be done as part of the current config conversion ticket https://jts.seagate.com/browse/EOS-17389 or can be addressed in a separate ticket later after EOS-17389 is resolved.

Reply:
Note: New ticket https://jts.seagate.com/browse/EOS-18174 created to track this task in the next sprints